File: Congruence/Java/LinearCongruence.java
/*
Name - Nikhil Ranjan Nayak
Regd no - 1641012040
Desc - Linear Congruence Solver
*/
import java.util.Scanner;

public class LinearCongruence {

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int a, c, m, g;
		System.out.println("a ≅ c(mod m)");
		System.out.println("a - ");
		a = sc.nextInt();
		System.out.println("c - ");
		c = sc.nextInt();
		System.out.println("m - ");
		m = sc.nextInt();
		System.out.println(a + " ≅  " + c + "(mod " + m + ")");
		g = gcd(a, m); 
		System.out.println("GCD - " + g);
		if(m % g == 0)
		{
			for(int x = 0; x < m; x++)
			{
				if((a*x) % m == c)
					System.out.println("\nx ≅ "+ x + "(mod " + m + ")");
			}
			System.out.println("");
		}
	}

	public static int gcd(int a,int b) {
		if(a == 0)
        		return b;
    		return gcd(b%a, a);
    	}
}

File: Congruence/C/LinearCongruence.c
/*
Name - Nikhil Ranjan Nayak
Regd no - 1641012040
Desc - Linear Congruence Solver
*/
#include "stdio.h"
int gcd(int, int);
void main()
{
	int x, a, c, m, g;
	printf("\n a ≅ c(mod m)");
	printf("\n a - ");
	scanf("%d", &a);
	printf("\n c - ");
	scanf("%d", &c);
	printf("\n m - ");
	scanf("%d", &m);
	printf("\n %d ≅ %d(mod %d)", a, c, m);
	g = gcd(a, m);
	printf("\n GCD - %d", g);
	if(m % g == 0)
	{
		for(x = 0; x < m; x++)
		{
			if((a*x) % m == c)
				printf("\n x ≅ %d(mod %d)", x, m);
		}
		printf("\n");
	}
}

int gcd(int a, int b)
{
	if(a == 0)
        	return b;
    	return gcd(b%a, a);
}
